src/Style.hs

```diff
@@ -283,12 +269,40 @@ attribute = identifier -- TODO: Naming convention - same as identifiers?
 -- styPrettyPrint b = ""
 
 --------------------------------------------------------------------------------
--- Functions used by "Runtime"
+-- Functions used by "Runtime" -- Style Translator
+-- Translates from a Style AST to:
+-- * a list of geometric primitives
+-- * a list of constraints
+-- * a list of objectives
+
+-- Type aliases for readability in this section
+-- | 'StyContext' maintains the current output of the translater
+type StyContext a =
+    (StyDict,         -- | dictionary mapping Substance ID to Style objects
+    [ObjFnInfo a],    -- | List of objective functions
+    [ConstrFnInfo a]) -- | List of constraints
```

Comment:
should we also build a list of computations in Style? right now i'm doing it in an inconsistent manner:
* since comps can only set properties of shapes, they are added to a shape's list of properties (config) in addSpec
* comps are discovered and argument names resolved in Runtime, in compsAndVars (called by getShape)